doc: Extension Function (Kotlin) (#3666)
Codecademy/docs: content/kotlin/concepts/functions/extension-function.md

> ## Extension Function 

Reviewer: ```suggestion
## Extension Functions 
```

> - These functions provide ease to add more functionality to the existing classes, without inheriting them. This is called an Extension Function. 

Reviewer: ```suggestion
Kotlin provides the opportunity to add more functionality to existing classes, by leveraging available methods without having to inherit the class. These declarations that use existing methods in new or custom classes are called extension functions. 
```

> - Below example demonstrates the use of this concept : 

Reviewer: In the other sections of this entry the author has done a good job of explaining the different elements and options for the syntax. That's missing here, there should be more information about how extension functions work. Instead of just presenting an example there should be details about what are the key syntax elements, and any other information that would be helpful.

```

// Define an extension function for the Int class
fun Int.square(): Int {
    return this * this
}

fun main() {
    val number = 5
    
    // Calling the extension function to calculate the square of the number
    val square = number.square()
    
    println("The square of $number is $square")
}

```

> - Output : The square of 5 is 25

Reviewer: The other examples have the output commented within the code

> ## Extension Functions 

Reviewer: Originally, I incorrectly reviewed this as an edit to existing entry. But, this is being added as a completely new doc and the path should be -
content/kotlin/concepts/extension-functions/extension-functions.md

Please update the folder and filename accordingly

Reviewer: This is also missing the metadata section please use another entry as a reference or copy and edit the metadata from one of our templates
https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md?plain=1

> - Kotlin provides the opportunity to add more functionality to existing classes, by leveraging available methods without having to inherit the class. These declarations that use existing methods in new or custom classes are called extension functions. 

Reviewer: ```suggestion
Kotlin provides the opportunity to add more functionality to existing classes, by leveraging available methods without having to inherit the class. These declarations that use existing methods in new or custom classes are called extension functions. 
```

Reviewer: all these lines should not be bullet points

- Below example demonstrates the use of this concept : 

```

// Define an extension function for the Int class
fun Int.square(): Int {
    return this * this
}

fun main() {
    val number = 5
    
    // Calling the extension function to calculate the square of the number
    val square = number.square()
    
    println("The square of $number is $square")
}

Output : The square of 5 is 25
```

> ### Use of extension functions in a real-world project

Reviewer: I appreciate the interest in adding a real-world problem, but we don't generally link out to other outside pages. Please revise the example or delete it

Author: @caupolicandiaz It was told to add the links [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#links)
